### DISCUSSION TITLE
Generating configuration files for command line usage from GUI

### DISCUSSION THREAD

**Discussion by JoshuaWard201** [HEXRD/hexrdgui#1722] [Q&A]
Hey everyone, I was curious if anyone has ever used the GUI to generate configuration files that can then be used in the CLI. The configuration files I have been using essentially outline the thresholds and tolerances, everything you click now in the gui when running analysis. My hope is that I can look at one data set in the GUI and get everything the way I want it, then generate the configuration file, that I can then edit for every other dataset and bash script it through the CLI. Any thoughts?

**Comment by JoshuaWard201** [comment]
@psavery , any thoughts. Sorry I forgot to put this in the actual body.

**Reply by psavery** [reply]
Hey Josh! We do actually have something like this in place already for the full HEDM workflow. If you run through indexing and fit grains in the GUI, then in the fit grains results dialog, there is an "Export Workflow" button where you pick a directory, and it writes out all of the files needed to run it with the `hexrd` command line interface. That way, you can do it once in the GUI and then run it in batch on everything else. See [here](https://hexrdgui.readthedocs.io/en/latest/hedm/fit_grains/#export-full-workflow) for more details.

Are you doing the full HEDM workflow?

**Reply by JoshuaWard201** [reply]
Hi @psavery , I am doing the full HEDM workflow

**Reply by psavery** [reply]
Okay! See that documentation in the link above, then. Hopefully that will be what you need!
